plugins/org.yakindu.sct.model.resource/src/org/yakindu/sct/model/resource/tasks/DomainSpecificTaskFinder.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2017 committers of YAKINDU and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * Contributors:
+ * 	committers of YAKINDU - initial API and implementation
+ */
+package org.yakindu.sct.model.resource.tasks;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.xtext.EcoreUtil2;
+import org.eclipse.xtext.tasks.ITaskFinder;
+import org.eclipse.xtext.tasks.Task;
+import org.yakindu.sct.domain.extension.DomainRegistry;
+import org.yakindu.sct.domain.extension.IDomain;
+import org.yakindu.sct.model.sgraph.SGraphPackage;
+import org.yakindu.sct.model.sgraph.resource.AbstractSCTResource;
+/**
+ * 
+ * @author Johannes Dicks - Initial contribution and API
+ * 
+ */
+@SuppressWarnings("restriction")
+public class DomainSpecificTaskFinder implements ITaskFinder {
```

Comment:
thought about the same but decided to just implement ITaskFinder to avoid the need of additional bindings in org.yakindu.sct.model.resource.SCTExecutableExtensionFactory.